src/main/java/net/openhft/chronicle/queue/util/NamedTableStoreLock.java

```diff
@@ -0,0 +1,71 @@
+package net.openhft.chronicle.queue.util;
+
+import net.openhft.chronicle.bytes.ref.BinaryLongReference;
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.queue.impl.TableStore;
+
+import java.io.IOException;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+
+import static java.lang.String.format;
+
+public class NamedTableStoreLock {
+
+    private static final int PID = Jvm.getProcessId();
+    private static final long UNLOCKED = Long.MIN_VALUE;
+
+    private final String lockName;
+    private final BinaryLongReference lockLongReference;
+
+    public NamedTableStoreLock(TableStore<?> tableStore, String lockName) {
+        this.lockName = lockName;
+        this.lockLongReference = (BinaryLongReference) tableStore.doWithExclusiveLock(ts -> ts.acquireValueFor(lockName, UNLOCKED));
+    }
+
+    public FileLock lock() {
+        return lock(lockLongReference, lockName);
+    }
+
+    public static FileLock lock(BinaryLongReference binaryLongReference, String lockName) {
+        FileLock lock;
+        try {
+            lock = binaryLongReference.lock(false);
+            if (lock != null) {
+                // This process acquired the replicator lock
+                if (binaryLongReference.getVolatileValue() != PID) {
+                    // Update the value to reflect who is holding the lock
+                    binaryLongReference.setVolatileValue(PID);
+                }
+            } else {
+                Jvm.warn().on(NamedTableStoreLock.class, format("Couldn't acquire lock %s, it is currently held by %d",
+                        lockName, binaryLongReference.getVolatileValue()));
+            }
+            return lock;
+        } catch (IOException e) {
+            Jvm.error().on(NamedTableStoreLock.class, String.format("Error attempting to acquire lock %s", lockName), e);
+            return null;
+        }
+    }
+
+    public boolean isLocked() {
+        return isLocked(lockLongReference, lockName);
+    }
+
+    public static boolean isLocked(BinaryLongReference lockReference, String lockName) {
+        try {
+            // When testing isLocked try and acquire a shared lock so we don't mistake
+            // others testing isLocked for the actual holder
+            final FileLock fileLock = lockReference.tryLock(true);
+            if (fileLock != null) {
+                fileLock.release();
+            }
+            return fileLock == null;
+        } catch (OverlappingFileLockException e) {
+            return true;
+        } catch (IOException e) {
+            throw new IllegalStateException("Error checking isLocked " + lockName, e);
+        }
+    }
+}
+
```

Comment:
I'm not sure if I should instead give this a similar API to `WriteLock`. Its a little bit different to that, in that only the holder of the lock can unlock it (you need the FileLock object to do that).

Reply:
It would be a bit weird because only the INSTANCE that locked() the lock would be able to unlock, even if the process was the one holding the lock. Unless I store the lock in a static field on the class? Then you'd probably have to synchronize the lock and unlock methods.